app/models/_base.py

```diff
@@ -0,0 +1,3 @@
+class Computation:
```

Comment:
1. why not using https://andrewbrookins.com/technology/building-implicit-interfaces-in-python-with-protocol-classes/ ?
2. add typings

also, `compute` is the wrong abstraction. The types are different in `chat.py` and `matching.py`